components/config-provider/index.tsx

```diff
@@ -68,19 +70,31 @@ export interface ConfigProviderProps {
   dropdownMatchSelectWidth?: boolean;
 }
 
-const ConfigProvider: React.FC<ConfigProviderProps> & {
-  ConfigContext: typeof ConfigContext;
-} = props => {
-  React.useEffect(() => {
-    if (props.direction) {
-      message.config({
-        rtl: props.direction === 'rtl',
-      });
-      notification.config({
-        rtl: props.direction === 'rtl',
-      });
-    }
-  }, [props.direction]);
+interface ProviderChildrenProps extends ConfigProviderProps {
+  parentContext: ConfigConsumerProps;
+  legacyLocale: Locale;
+}
+
+const ProviderChildren: React.FC<ProviderChildrenProps> = props => {
```

Comment:
Why create another FC to handle this?

Reply:
Current implementation use a fuction to render children, where parent context's config is passed in.

Since we can't use Hooks inside a nested function. It has to be another component.